scripts/mainmenu.gd

```diff
@@ -50,6 +51,80 @@ func _process_stage6_sex_options():
 		get_node("TextureFrame/newgame/stage6/balls").set_disabled(true)
 		get_node("TextureFrame/newgame/stage6/balls").add_item('none')
 
+
+#QMod - Renamed, tweaked variable initialization
+func _ready_newgame_creator():
+	#Connect UI
+	#Connect stage selection buttons
+	for i in get_node("TextureFrame/newgame/stagespanel/VBoxContainer").get_children():
+		if i.get_name() != 'cancel':
+			i.connect("pressed", self, '_select_stage', [i])
+	
+	#Connect text entry boxes	
+	for i in get_tree().get_nodes_in_group("lookline"):  
+		i.connect("text_changed", self, '_lookline_text', [i])
+	
+	#Connect list options
+	for i in get_tree().get_nodes_in_group("lookoption"):  
+		i.connect("item_selected", self, '_option_select', [i])
+	
+	#Connect stat up/down buttons	
+	for i in get_tree().get_nodes_in_group("statup"):  
+		i.connect("pressed",self,'statup',[i])		
+	for i in get_tree().get_nodes_in_group("statdown"):
+		i.connect("pressed",self,'statdown',[i])
+	
+	#Connect slave name entry boxes
+	for i in ["TextureFrame/newgame/stage8/slavename", "TextureFrame/newgame/stage8/slavesurname"]:
+		var temp = get_node(i)
+		temp.connect("text_changed", self, '_slavename_text', [temp])
+
+	#Connect slave customization options
+	for i in get_tree().get_nodes_in_group("slaveoption"):  
+		i.connect("item_selected",self,'_slave_option', [i])
+	
+	#Connect game options/settings
+	for i in get_tree().get_nodes_in_group("startoption"):  
+		i.connect("pressed",self,'_option_toggle',[i])
+	
+	#Connect virgin option
+	get_node("TextureFrame/newgame/stage6/virgin").connect("pressed", self, '_virgin_press')
+	
+	#Initialize newgame variables
+	player = globals.newslave(playerDefaults.race, playerDefaults.age, playerDefaults.sex, playerDefaults.origins) #Prefer to use a constructor/builder
+	#ralph
+	player.playercleartraits()
+	#/ralph
+	player.hairstyle = 'straight'
+	player.beautybase = variables.playerstartbeauty
+	
+	startSlave = globals.newslave(slaveDefaults.race, slaveDefaults.age, slaveDefaults.sex, slaveDefaults.origins) #Prefer a constructor/builder
+	startSlave.cleartraits()
+	startSlave.beautybase = variables.characterstartbeauty
+	startSlave.memory = slaveBackgrounds.back()
+	
+	globals.resources.panel = null #Clear global variables
+	globals.showalisegreet = false
+
+	_build_player_portraits() #Build Player portrait list
+
+func regenerateplayer():
+	var imageportait = player.imageportait
+	player = globals.newslave(player.race, player.age, player.sex, 'slave')
+	globals.player = player
+	#ralph
+	player.playercleartraits()
```

Comment:
Im checking with Ralph what the purpose of player.playercleartraits() is for. Wondering if its a toggle thing or not.

Reply:
Fixed